ui2/src/pages/login/QuickSetup.tsx

```diff
@@ -0,0 +1,560 @@
+} from '@daml.js/da-marketplace/lib/Marketplace/Trading/Role';
+import {
+  Offer as MatchingOffer,
+  Service as MarchingService,
+} from '@daml.js/da-marketplace/lib/Marketplace/Trading/Matching/Service';
+
+import { CreateEvent } from '@daml/ledger';
+
+type Offer = CustodianOffer | DistributorOffer | SettlementOffer | ExchangeOffer | MatchingOffer;
+
+enum serviceOptionsEnum {
+  CUSTODY = 'Custody',
+  TRADING = 'Trading',
+  MATCHING = 'Matching',
+  SETTLEMENT = 'Settlement',
+  LISTING = 'Listing',
+}
+
+interface IServiceContractSetupData {
+  party?: PartyDetails;
+  service?: string;
+}
+
+const services = ['Custody', 'Trading', 'Matching', 'Settlement', 'Listing'];
+
+const OfferServiceContractSetup = (props: {
+  credentials: Credentials;
+  serviceSetupData?: IServiceContractSetupData;
+  setServiceSetupData: (data: IServiceContractSetupData) => void;
+  parties: PartyDetails[];
+  creatingRoleContracts: boolean;
+  onComplete: () => void;
+}) => {
+  const {
+    credentials,
+    serviceSetupData,
+    setServiceSetupData,
+    parties,
+    onComplete,
+    creatingRoleContracts,
+  } = props;
+
+  const [status, setStatus] = useState<string>();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [marketSetupDataMap, setMarketSetupDataMap] = useState<Map<string, string[]>>(new Map());
+
+  const ledger = useLedger();
+  const operator = useWellKnownParties().parties?.userAdminParty || 'Operator';
+
+  const custodianRoles = useStreamQueries(CustodianRole);
+  const exchangeRoles = useStreamQueries(ExchangeRole);
+  const distributorRoles = useStreamQueries(DistributorRole);
+  const settlementServices = useStreamQueries(SettlementService);
+  const matchingServices = useStreamQueries(MarchingService);
+  const operatorService = useStreamQueries(OperatorService);
+
+  const selectedParty = serviceSetupData?.party;
+  const selectedService = serviceSetupData?.service;
+
+  useEffect(() => {
+    setLoading(
+      custodianRoles.loading ||
+        distributorRoles.loading ||
+        settlementServices.loading ||
+        exchangeRoles.loading ||
+        matchingServices.loading ||
+        operatorService.loading
+    );
+  }, [
+    custodianRoles.loading,
+    distributorRoles.loading,
+    settlementServices.loading,
+    exchangeRoles.loading,
+    matchingServices.loading,
+    operatorService.loading,
+  ]);
+
+  useEffect(() => {
+    if (loading) {
+      return;
+    }
+
+    let newMarketData: Map<string, string[]> = new Map();
+
+    custodianRoles.contracts.forEach(c => addMarketData(c.payload.provider, 'Custody'));
+    distributorRoles.contracts.forEach(c => addMarketData(c.payload.provider, 'Listing'));
+    settlementServices.contracts.forEach(c => addMarketData(c.payload.provider, 'Settlement'));
+    exchangeRoles.contracts.forEach(c => addMarketData(c.payload.provider, 'Trading'));
+    matchingServices.contracts.forEach(c => addMarketData(c.payload.provider, 'Matching'));
```

Comment:
we should use the enum values instead of string literals, to keep things in one spot

Reply:
ah yes thanks